### PR TITLE
Formatter: preserve or emit canonical blank lines within method bodies (BT-987)

### PR DESCRIPTION
**Pull request by jamesc**
## Summary

Preserves author-placed blank lines within method bodies, block bodies, and module-level expressions during formatting. Previously the formatter stripped all blank lines within bodies since the AST didn't track them.

**Linear:** https://linear.app/beamtalk/issue/BT-987

### Key changes
- Added `preceding_blank_line: bool` field to `ExpressionStatement` AST node
- Added `Token::has_preceding_blank_line()` to detect 2+ newlines in leading trivia
- Parser sets the field in method bodies, block bodies, and module-level expressions
- Unparser emits an extra blank line before marked statements
- Consecutive blank lines are collapsed to a single canonical blank line

### Tests added
- Token blank line detection (5 cases: no newline, single, double, triple, split trivia)
- Method body blank line emission (with and without blank lines)
- Round-trip idempotency tests (with blank lines, without, consecutive collapse)
- Module-level blank line idempotency
- 65 parser snapshots updated with new `preceding_blank_line` field

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Parser now preserves author-placed blank lines before statements; formatting/emission retains those blank lines in modules, methods and blocks.

* **Tests**
  * Added tests covering blank-line preservation, idempotent round-trips, consecutive-blank-line behavior, and edge cases to ensure formatting stability.

### CODE REVIEW

File: crates/beamtalk-core/src/unparse/mod.rs [hunk 1]
```diff
@@ -1431,6 +1487,53 @@ mod tests {
...
+    fn consecutive_blank_lines_collapsed() {
+        // Multiple blank lines should collapse to a single one
+        let source = "Object subclass: Foo\n\n  doStuff =>\n    x := 1\n\n\n\n    x + 1\n";
+        let pass1 = unparse_module(&parse_source(source));
+        // After formatting, should have a blank line (not 3 blank lines)
+        // The blank line between statements is preserved
+        assert!(
+            pass1.contains("x := 1\n"),
+            "Expected blank line preserved, got:\n{pass1}"
+        );
```
[the remark this answers is not in the document]
Comment (jamesc) [reply crates/beamtalk-core/src/unparse/mod.rs]: Fixed in 6eaec09c — removed the misleading assertion. The blank line count is now verified solely by the `newline_count` check below.
